src/shell_manager.hh

```diff
@@ -34,6 +34,8 @@ class ShellManager : public Singleton<ShellManager>
 public:
     ShellManager(ConstArrayView<EnvVarDesc> builtin_env_vars);
 
+    static String find_shell();
```

Comment:
I don't think this function should be static, or even part of the `ShellManager` class. You could just leave the implementation in `shell_manager.cc`, and write down a prototype in `shell_manager.hh`.